### PR TITLE
[MCIAVI32] Media Player: Don't open empty files

### PR DESCRIPTION
**Pull request by atharva1910**
Opening Empty files in Mediaplayer leads to a crash
JIRA issue: [CORE-18669](https://jira.reactos.org/browse/CORE-18669)

### CODE REVIEW

**Review by KRosUser2 — APPROVED** [review]

**Comment by KRosUser2** [comment]
Lgtm. Please check if proposed patch is not needed anyway for the sake of robustness against Exception

**Comment by katahiromz** [comment]
It don't open zero-byte-sized files.
Then, how is an one-byte-sized file?

**Comment by atharva1910** [comment]
> It don't open zero-byte-sized files. Then, how is an one-byte-sized file?

I'm not sure I understand the question. 
The file is basically just an AVI header and no payload. So it contains no playable data

File: dll/win32/mciavi32/mmoutput.c [hunk 1]
```diff
@@ -398,6 +398,12 @@ BOOL MCIAVI_GetInfo(WINE_MCIAVI* wma)
 
 	mmioAscend(wma->hFile, &mmckInfo, 0);
     }
+
+    /* Empty file*/
```
Comment (katahiromz) [inline dll/win32/mciavi32/mmoutput.c:402]:
```suggestion
    /* Empty file */
```

Comment (atharva1910) [reply dll/win32/mciavi32/mmoutput.c]: Done 👍

**Comment by KRosUser2** [comment]
> It don't open zero-byte-sized files. Then, how is an one-byte-sized file?

@katahiromz : I don't get your point. We don't talk about 0 byte file here but 0 FRAME file.

File: dll/win32/mciavi32/mmoutput.c [hunk 2]
```diff
@@ -398,6 +398,12 @@ BOOL MCIAVI_GetInfo(WINE_MCIAVI* wma)
 
 	mmioAscend(wma->hFile, &mmckInfo, 0);
     }
+
+    /* Empty file */
+    if (alb.numVideoFrames == 0) {
+        return FALSE;
```
Comment (HBelusca) [inline dll/win32/mciavi32/mmoutput.c:404]: Maybe add a WARN trace?

**Comment by HBelusca** [comment]
Shouldn't this fix be sent upstream to Wine?

File: dll/win32/mciavi32/mmoutput.c (hunk 2 above)
Comment (KRosUser2) [reply dll/win32/mciavi32/mmoutput.c:404]: Good idea, same for the other place too (was the idea of the initial patch btw)

Comment (atharva1910) [reply dll/win32/mciavi32/mmoutput.c:405]: Done :+1:

**Comment by atharva1910** [comment]
> Shouldn't this fix be sent upstream to Wine?

As I understand, this is not part of the Wine Win32 DLLs but rather the MediaPlayer which is a part of ReactOS

**Review by learn-more — APPROVED** [review]

**Comment by HBelusca** [comment]
> > Shouldn't this fix be sent upstream to Wine?
> 
> As I understand, this is not part of the Wine Win32 DLLs but rather the MediaPlayer which is a part of ReactOS

I don't know; I see Wine authors listed at the beginning of the file, as well as its presence there:
https://github.com/wine-mirror/wine/tree/master/dlls/mciavi32
as well as the fact it's mentioned it's synced with Wine there (though a bit outdated): https://github.com/reactos/reactos/blob/master/media/doc/WINESYNC.txt#L95

**Review by HBelusca — APPROVED** [review]

**Pull request merged**